From b2822dbc25966a1a42fcebf8ea69ced25abc465b Mon Sep 17 00:00:00 2001
From: kfb77 <>
Date: Tue, 19 May 2020 23:34:16 +0200
Subject: [PATCH 2/2] add original VDR EventID to Event Aux xml

---
 update.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/update.c b/update.c
index ee2d5d0..97bdeb4 100644
--- a/update.c
+++ b/update.c
@@ -1873,6 +1873,14 @@ cEvent* cUpdate::createEventFromRow(const cDbRow* row)
                xml.appendElement(auxFields[i], value->getIntValue());
          }
 
+         // add original vdr EventID to Aux field
+         if (row->hasValue("SOURCE", "vdr"))
+         {
+             int eventid = (int) row->getBigintValue("EVENTID");
+             tell(3, "cUpdate::createEventFromRow(): new event for '%s' epg2vdr EventID %i vdr EventID %i", e->Title(), e->EventID(), eventid);
+             if (eventid) xml.appendElement("vdrEventID", eventid);
+         }
+
          // finally add some fields of the view
 
          xml.appendElement("source", viewMergeSource->getStrValue());
-- 
2.17.1

